### PR TITLE
docs/contribution.md: Update required Go version 

### LINKED ISSUE #5532: Update the go version in docs
In CONTRIBUTING.md in [prerequisites](https://github.com/thanos-io/thanos/blob/main/CONTRIBUTING.md#prerequisites) section Go version is set as 1.13.9 since thanos is using 1.18 version

### PR DESCRIPTION
**Pull request by Akshit42-hue**
fixes #5532 

Updated Go version in [prerequisites](https://github.com/thanos-io/thanos/blob/main/CONTRIBUTING.md#prerequisites)

### CODE REVIEW

**Comment by Akshit42-hue** [comment]
Please take a look @saswatamcode @matej-g @yeya24

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ The following section explains various suggestions and procedures to note during
 ### Prerequisites
 
 * It is strongly recommended that you use Linux distributions systems or OSX for development.
-* Go 1.13.9 or newer installed.
```
Comment (douglascamata) [inline CONTRIBUTING.md:70]: We're not using at Go 1.18.x only feature. According to our `go.mod` file, Go has to be 1.17.x or higher.

**Comment by Akshit42-hue** [comment]
PTAL @douglascamata

**Review by saswatamcode — APPROVED** [review]
Thanks!

Can you run `make docs` and commit to fix CI? 🙂

**Comment by Akshit42-hue** [comment]
> Thanks!
> 
> Can you run `make docs` and commit to fix CI? slightly_smiling_face

On it, Thanks for the review :)

**Comment by Akshit42-hue** [comment]
The CI is still failing can someone please help me out :)

**Comment by douglascamata** [comment]
@Akshit42-hue push an empty commit and let it rerun. There are some intermittent failures sometimes in the dead link check of docs due to timeouts.

**Review by douglascamata — APPROVED** [review]
LGTM :shipit:

**Comment by Akshit42-hue** [comment]
> @Akshit42-hue push an empty commit and let it rerun. There are some intermittent failures sometimes in the dead link check of docs due to timeouts.



> @Akshit42-hue push an empty commit and let it rerun. There are some intermittent failures sometimes in the dead link check of docs due to timeouts.

Done that but nothing gets changed

**Comment by douglascamata** [comment]
@Akshit42-hue I don't see an empty commit pushed since the last failure. Are you sure you pushed something?

**Comment by Akshit42-hue** [comment]
> @Akshit42-hue I don't see an empty commit pushed since the last failure. Are you sure you pushed something?

Yes, but I am trying to figure out how can an empty commit be push
Screenshot from 2022-08-01 20-26-44

**Comment by douglascamata** [comment]
@Akshit42-hue you can create an empty commit with `git commit -s --allow-empty -m "Retrigger CI"`

**Comment by Akshit42-hue** [comment]
> @Akshit42-hue you can create an empty commit with `git commit -s --allow-empty -m "Retrigger CI"`

All the test have been passed. Thank you! @douglascamata for helping me out 😀

**Review by squat — APPROVED** [review]
Thanks @Akshit42-hue

**Pull request merged**